2996_reflection/reflection.md

```diff
@@ -4163,33 +4182,34 @@ consteval void g(std::meta::info r, X<false> xv) {
 
-[#]{.pnum} A `$reflect-expression$` of the form `^^ $nested-name-specifier$@~_opt_~@ $template-name$` represents the primary class template, function template, primary variable template, or alias template denoted by the `$template-name$`.
+- [#.#]{.pnum} Otherwise, if the `$identifier$` names a type alias that was introduced by the declaration of a template parameter, `$R$` represents the underlying entity of that type alias. For any other `$identifier$` that names a type alias, `$R$` represents that type alias.
+- [#.#]{.pnum} Otherwise, if the `$identifier$` is a `$class-name$` or an `$enum-name$`, `$R$` represents the denoted type.
+- [#.#]{.pnum} Otherwise, the `$qualified-reflection-name$` shall be an `$id-expression$` `$I$` and `$R$` is `^^ $I$` (see below).
 
-[#]{.pnum} A `$reflect-expression$` of the form `^^ $nested-name-specifier$@~_opt_~@ $concept-name$` represents the concept denoted by the `$concept-name$`.
+[#]{.pnum} A `$reflect-expression$` of the form `^^ $type-id$` represents an entity determined as follows:
 
-[#]{.pnum} A `$reflect-expression$` of the form `^^ $type-id$` is interpreted as follows:
+- [#.#]{.pnum} If the `$type-id$` names a type alias that is a specialization of an alias template, `$R$` represents that type alias.
+- [#.#]{.pnum} Otherwise, `$R$` represents the type denoted by the `$type-id$`.
 
-* [#.#]{.pnum} If the `$type-id$` is a `$typedef-name$` that was introduced by the declaration of a template parameter, the `$reflect-expression$` represents the type denoted by that `$typedef-name$`.
-* [#.#]{.pnum} Otherwise, if the `$type-id$` is any other `$typedef-name$`, the `$reflect-expression$` represents the type alias associated with that `$typedef-name$`.
-* [#.#]{.pnum} Otherwise, the `$reflect-expression$` represents the type denoted by the `$type-id$`.
+[All other cases for `^^ $type-id$` are covered by `$qualified-reflection-name$`.]{.ednote}
 
-[#]{.pnum} A `$reflect-expression$` of the form `^^ $id-expression$` is interpreted as follows:
+[#]{.pnum} A `$reflect-expression$` `$R$` of the form `^^ $id-expression$` represents an entity determined as follows:
 
-  * [#.#]{.pnum} If the `$id-expression$` denotes an overload set `$S$` and overload resolution for the expression `&$S$` determines a unique function `$F$` ([over.over]{.sref}), the `$reflect-expression$` represents `$F$`.
+  * [#.#]{.pnum} If the `$id-expression$` denotes an overload set `$S$` and overload resolution for the expression `&$S$` with no target selects a unique function ([over.over]{.sref}), `$R$` represents that function.
```

Comment:
Would it make sense to say something like:

> If the `$id-expression$` denotes an overload set `$S$`, then overload resolution for the expression `&$S$` with no target shall select a unique function; ([over.over]{.sref}), `$R$` represents that function.
Comment

Just make it more clear that if we actually have multiple overloads that we're ill-formed (i.e. there's not some later bullet that catches this case)?